### PR TITLE
App Submission: Seafile

### PR DESCRIPTION
**Pull request by IMPranshu**
# App Submission

### SeaFile
...

### 256x256 SVG icon
https://svgshare.com/i/vV3.svg
...

### Gallery images
Screenshot from 2022-11-14 14-31-56
Screenshot from 2022-11-14 14-32-06
Screenshot from 2022-11-14 14-32-19
Screenshot from 2022-11-14 14-32-36
Screenshot from 2022-11-14 14-32-44

...

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [x] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

**Comment by nmfretz** [comment]
Awesome @IMPranshu, this has been requested in the past and is quite popular. I will review soon!

**Comment by nmfretz** [comment]
Hey @IMPranshu, it looks like this might still be a work-in-progress so I have converted this PR to a Draft while you are working on it and testing. 

Here's a few things to check that I noticed at first glance:
- The Umbrel `app_proxy` container is missing from the `docker-compose.yml`. Was this intentional?
- seafile is running as root (may not be necessary).
- The images are quite old (~1yr).
- Can you please check if we can remove the custom `seafile-net` network and custom `container_name`'s, so that we simplify the yml and let Docker Compose automatically handle this? I think everything should still work correctly.
- The data directories for the volume bind mounts are missing from this repo.

Let me know when you have this working nicely and I'll review and test!

**Comment by swarupbhc** [comment]
Hi, @nmfretz please review this.

File: seafile/umbrel-app.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+manifestVersion: 1
+id: seafile
+category: Files
```
Comment (ParthJadhav) [inline seafile/umbrel-app.yml:3]: Hey @IMPranshu , due to a recent change the category values should be lowercase. 

```suggestion
category: files
```

**Review by ParthJadhav — CHANGES_REQUESTED** [review]
Tested on Umbrel-dev server. Works fine. I didn't test further because I didn't know where to create an account to test it.

there are couple of changes which I've mentioned.

Thanks for the contribution ❤️

**Comment by johnpc** [comment]
This is awesome! 

However it does require that you set these manually to the host-ip:port after installation, otherwise all uploads fail: 


Additionally, the username/password of `me@example.com` and `asecret` are not conventional for umbrel. I think they should be changed (possibly `umbrel` and `moneyprintergobrrr` instead).

Finally, it is frustrating for me that the drive folder is nested under `app-data`. This means that if the app is deleted, all files uploaded via seafile are destroyed. I would prefer if it pointed to the data volume like ` - ${UMBREL_ROOT}/data/storage/seafile-data:/shared` for example.

Seafile is an excellent program and it certainly belongs in the umbrel app store. I hope this PR gets merged

**Comment by nmfretz** [comment]
Alrighty, I went through this today and got Seafile working https://github.com/getumbrel/umbrel-apps/pull/676/commits/dfb9b366a5e640c2f61a26388f8324f738057954. This is pretty much ready to go! @IMPranshu @smolgrrr @johnpc may I please get one of you to test as well?

@johnpc:

> Is it possible to set those values to the correct ones by default using environment variables?

Yep, I set env vars so that uploading works out-of-the-box after installing as long as you upload from the UI using your main hostname... so `umbrel.local` for most people.

image



Users who want to customize this can do so in the UI and it will persist.


> Additionally, the username/password of me@example.com and asecret are not conventional for umbrel. I think they should be changed (possibly umbrel and moneyprintergobrrr instead).

Good call. I have changed the username to `umbrel` and the default password to a unique, deterministic one as we do for other apps. These values will be available to users in the app store when they install the app.

> Finally, it is frustrating for me that the drive folder is nested under app-data. This means that if the app is deleted, all files uploaded via seafile are destroyed. I would prefer if it pointed to the data volume like  - ${UMBREL_ROOT}/data/storage/seafile-data:/shared for example.

Ya, I hear you. The reason we do this is because most users would expect deleting an app to also delete all data associated with it (especially if they are doing so to free up room on their device). Let me chat with the team and get back to you on this.

**Comment by IMPranshu** [comment]
Tested this on my Pi, working 💪

**Comment by nmfretz** [comment]
Thanks @IMPranshu!

I added some additional instructions to the app description. I'll merge this once we have the gallery assets ready.

**Comment by IMPranshu** [comment]
😎

**Comment by nmfretz** [comment]
Gallery assets are now ready! Going live to the app store 🎉

https://github.com/getumbrel/umbrel-apps-gallery/tree/master/seafile
image

**Pull request merged**